ports/stm/common-hal/analogio/AnalogIn.c

```diff
@@ -31,10 +31,49 @@
 
 #include "shared-bindings/microcontroller/Pin.h"
 
+#ifdef STM32F405xx 
```

Comment:
Does the LL use prevent you from using `#include STM32_HAL_H` here (that's a macro for `<stm32$(MCU_SERIES_LOWER)xx_hal.h>`)? It's been a little while so I don't remember if LL files get included with that. if `STM32_HAL_H` doesn't work, please simplify these if statements to use `CPY_STM32F4`, `CPY_STM32F7`, `CPY_STM32H7` instead of repeating for every MCU variant.